script/D3MDeploy.s.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-FileCopyrightText: © 2022 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.0;
+
+import "forge-std/Script.sol";
+import { stdJson } from "forge-std/StdJson.sol";
+import { MCD, DssInstance } from "dss-test/MCD.sol";
+import { ScriptTools } from "dss-test/ScriptTools.sol";
+
+import {
+    D3MDeploy,
+    D3MInstance
+} from "../src/deploy/D3MDeploy.sol";
+
+contract D3MDeployScript is Script {
+
+    using stdJson for string;
+    using ScriptTools for string;
+
+    string config;
+    DssInstance dss;
+
+    string d3mType;
+    address admin;
+    address hub;
+    bytes32 ilk;
+    D3MInstance d3m;
+
+    function run() external {
+        config = ScriptTools.loadConfig("aave");
```

Comment:
Do we want to fallback here or should we actually error if there isn't a config fle env variable?

Reply:
Yeah this is a good point. I also have an issue with automatically defaulting to mainnet here which actually messed me up when I deployed the Hub to Goerli yesterday. I think I like having these defaults for integration tests, but not for deploying.

Maybe another function on `ScriptTools` to indicate that we everything explicitly defined or error? What do you think?

Reply:
You mean to have an alternative `loadConfig`?

Reply:
Perhaps. I also want to have it revert if the CHAINID is not specified instead of just assuming mainnet.

Reply:
I think we should follow one or the approach but not really having both options, as it will end up being confusing. I'm fine either way but if you already got an experience of getting messed up due to the fallback of the chain id, then might make sense to enforce always to define it.

Reply:
Ok. How about an overload on `loadConfig()` with empty params which will revert if nothing is defined and we axe the default of chainid = 1 and always require it specified even for integration tests.

Reply:
Ok perfect, sounds good. I've approved that PR.

Reply:
updated